pkg/controller/main-controller.go

```diff
@@ -1316,6 +1317,29 @@ func (c *Controller) syncHandler(key string) (Result, error) {
 			// return nil so we don't re-queue this work item, this error won't get fixed by reprocessing
 			return WrapResult(Result{}, nil)
 		}
+
+		// check if the system config has changed
+		// if changed, minio request the systemCfg must be the same to restart.
+		expectSystemCfg, err := c.getSystemCfgFromStatefulSet(ctx, expectedStatefulSet)
+		if err != nil {
+			return WrapResult(Result{}, err)
+		}
+		existSystemCfg, err := c.getSystemCfgFromStatefulSet(ctx, existingStatefulSet)
+		if err != nil {
+			return WrapResult(Result{}, err)
+		}
+		if !maps.Equal(expectSystemCfg, existSystemCfg) {
```

Comment:
you can also use `set.StringSet` from minio-go here to use things like `.Difference().IsEmpty()`

Reply:
ah I see you want differentiate based on the map values as well